gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/SubscriptionInfo.java

```diff
@@ -108,31 +157,60 @@ public SubscriptionInfo build() {
     ackDeadlineSeconds = builder.ackDeadlineSeconds;
   }
 
+  /**
+   * Returns the name of the topic this subscription refers to.
+   */
   public String topic() {
     return topic;
   }
 
+  /**
+   * Sets the name of the subscription. The name must start with a letter, and contain only
+   * letters ({@code [A-Za-z]}), numbers ({@code [0-9]}), dashes ({@code -}), underscores
+   * ({@code _}), periods ({@code .}), tildes ({@code ~}), plus ({@code +}) or percent signs
+   * ({@code %}). It must be between 3 and 255 characters in length and cannot begin with the
+   * string {@code goog}.
+   */
   public String name() {
     return name;
   }
 
+  /**
+   * Returns the push configuration for the subscription. If set, the subscription is in push mode
+   * and the returned value defines the push endpoint. If {@code null}, the subscription is in pull
+   * mode.
+   */
   public PushConfig pushConfig() {
     return pushConfig;
   }
 
+  /**
+   * Returns the maximum time after a subscriber receives a message before the subscriber should
+   * acknowledge the message. After message delivery but before the ack deadline expires and
+   * before the message is acknowledged, it is an outstanding message and will not be delivered
+   * again during that time (on a best-effort basis). For pull subscriptions, this value is used
+   * as the initial value for the ack deadline. To override the ack deadline value for a given
+   * message, use {@link PubSub#modifyAckDeadline(String, int, TimeUnit, Iterable)}. For push
+   * delivery, this value is used to set the request timeout for the call to the push endpoint. If
+   * not specified, the default value of 10 seconds is used.
+   */
   public long ackDeadlineSeconds() {
     return ackDeadlineSeconds;
   }
 
   @Override
-  public boolean equals(Object o) {
-    if (this == o) {
+  public boolean equals(Object obj) {
+    if (this == obj) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (obj == null || !obj.getClass().equals(this.getClass())) {
       return false;
     }
-    return Objects.equals(toPb(), ((SubscriptionInfo) o).toPb());
+    SubscriptionInfo other = (SubscriptionInfo) obj;
```

Comment:
Yes it works but now the `toPb()` method takes also the project id as a string. It seemed ugly to me to pass an empty string just for the purpose of comparing the objects